### ISSUE TITLE
When the response is rewritten, the set-cookie header multiplies.

### ISSUE THREAD

**Issue by tsctx** [honojs/hono#2566]
### What version of Hono are you using?

4.2.8

### What runtime/platform is your app running on?

Node

### What steps can reproduce the bug?

```js
import { Hono } from "hono";

const app = new Hono();

app.get("/", async (c) => {
  c.res = new Response(null, {
    headers: [
      ["set-cookie", "a"],
      ["set-cookie", "b"],
      ["set-cookie", "c"],
    ],
  });
  return c.text("Hi");
});
console.log([...(await app.request("/")).headers])
```

### What is the expected behavior?

```js
[
  [ 'content-type', 'text/plain; charset=UTF-8' ],
  [ 'set-cookie', 'a' ],
  [ 'set-cookie', 'b' ],
  [ 'set-cookie', 'c' ]
]
```

### What do you see instead?

```js
[
  [ 'content-type', 'text/plain; charset=UTF-8' ],
  [ 'set-cookie', 'a' ],
  [ 'set-cookie', 'b' ],
  [ 'set-cookie', 'c' ],
  [ 'set-cookie', 'a' ],
  [ 'set-cookie', 'b' ],
  [ 'set-cookie', 'c' ],
  [ 'set-cookie', 'a' ],
  [ 'set-cookie', 'b' ],
  [ 'set-cookie', 'c' ]
]
```

### Additional information

How to fix it?
https://github.com/honojs/hono/blob/347ef12a9b507f82aefef6b375ffb9f352da98e0/src/context.ts#L384-L393

Let's do it like this.

```js
 this.#res.headers.forEach((v, k) => { 
   if (k === 'set-cookie') { 
     this.#headers?.append(k, v)
   } else {
     this.#headers?.set(k, v)
   }
 }) 
```

**Comment by yusukebe** [comment]
@tsctx 

Fixed it and released the new version, v4.2.9, including the change. Thanks!

**Comment by ItsWendell** [comment]
Nice and thanks you both! This was an unintended side-effect of my previous patch, this fix is also a lot cleaner!
